src/cbmc_starter_kit/template-for-repository/proofs/lib/summarize.py

```diff
@@ -116,4 +147,19 @@ def print_proof_results(out_file):
 
 if __name__ == '__main__':
     args = get_args()
-    print_proof_results(args.run_file)
+    with_aws = args.cloudfront_domain and args.s3_uri
+    without_aws = not args.cloudfront_domain and not args.s3_uri
+    if not with_aws and not without_aws:
+        raise argparse.ArgumentTypeError(
```

Comment:
I don't think this is the right thing to do, this will dump a stack trace. `argparse` has [`exit()` and `error()`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.error) methods that print out the generated help text, plus your message to stderr and exit with 2, that's probably what you want here

Reply:
I think it's okay to raise `ArgumentTypeError` inside an argparse 'type' function, because argparse specifically catches that exception while processing the argument type. But it's not meant to be raised from arbitrary places in the code I think

Reply:
I personally prefer having both of these arguments in a subparser to make the condition that both should be called/not called in unison more obvious. However, after earlier offline chat, it was decided that a subparser wouldn't be used here

Reply:
I will be using `error()` as you pointed out.